setup.py

```diff
@@ -16,6 +16,28 @@
 except ImportError:
     py2exe = None
 
+
+class PackageNotFoundError(Exception):
+    pass
+
+
+# Check required non-python dependencies
+try:
+    import gi
+except ImportError as Err:
+    err_str = ("\n  ImportErrors:%s" % Err.message +
+               "\n\n  Non-Python dependency missing, please check the README.md file.")
+    raise PackageNotFoundError(err_str)
+
+try:
+    import pygtkcompat
+    pygtkcompat.enable()
+    pygtkcompat.enable_gtk(version='3.0')
+except ValueError as Err:
```

Comment:
We can combine this try block with the one above for gi and catch both types of exceptions.